File: Two pointer/125. 验证回文串/125. 验证回文串.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        l, r = 0, len(s)-1
        while l<r:
            while l<r and (self.check(s[l]) is None):
                l+=1
            while l<r and (self.check(s[r]) is None):
                r-=1
            print(l, r)
            if s[l].lower() != s[r].lower():
                return False
            l+=1
            r-=1
        return True
        
    def check(self, sub_s):
        if (ord("A") <= ord(sub_s) <= ord("Z")) or \
        (ord("a") <= ord(sub_s) <= ord("z")) or \
        (ord("0") <= ord(sub_s) <= ord("9")):
            return sub_s

File: Two pointer/125. 验证回文串/125. 验证回文串.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int l = 0;
        int r = s.size() - 1;
        while(l<r){
            while(l < r && isalnum(s[l]) == 0){
                l += 1;
            }
            while(l < r && isalnum(s[r]) == 0){
                r -= 1;
            }
            if(tolower(s[l]) != tolower(s[r])){
                return false;
            }
            l += 1;
            r -= 1;
        }
        return true;
    }

    // char check(char s){
    //     int k = s;//k為char s的10進位
    //     //0~9，A~Z，a~z
    //     if ((k >= 48 && k <= 57) or (k >= 65 && k <= 90) or (k >= 97 && k <= 122)){
    //         return s;
    //     }
    // }
};